ch00python/010exemplar.ipynb

```diff
@@ -245,37 +266,59 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "The Yandex API allows us to fetch a map of a place, given a longitude and latitude.\n",
-    "The URLs look like: https://static-maps.yandex.ru/1.x/?size=400,400&ll=-0.1275,51.51&z=10&l=sat&lang=en_US \n",
-    "We'll probably end up working out these URLs quite a bit. So we'll make ourselves another function to build up a URL given our parameters."
+    "We'll fetch a map of a place from the Google Maps server, given a longitude and latitude.\n",
+    "The URLs look like: https://mt0.google.com/vt?x=658&y=340&z=10&lyrs=s.    \n",
+    "The latitude & longitude need to be translated to the coordinate system used by Google Maps and we'll frequently be generating these URLs.\n",
+    "So we'll make ourselves another function to build up a request from the URL given our parameters."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 7,
-   "metadata": {},
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-08-06T13:18:41.903152Z",
+     "start_time": "2020-08-06T13:18:41.884801Z"
+    }
+   },
    "outputs": [],
    "source": [
+    "import os\n",
+    "import math\n",
     "import requests\n",
-    "def request_map_at(lat, long, satellite=True,\n",
-    "                   zoom=10, size=(400, 400)):\n",
-    "    base = \"https://static-maps.yandex.ru/1.x/?\"\n",
-    "  \n",
+    "\n",
+    "def request_map_at(lat, long, zoom=10, satellite=True):\n",
+    "    \n",
+    "    base_url = \"https://mt0.google.com/vt?\"\n",
+    "    \n",
+    "    gmaps_tile_size = 256\n",
+    "    num_tiles = 1 << zoom\n",
+    "    \n",
+    "    x_coord = (gmaps_tile_size / 2 + lat * gmaps_tile_size / 360.0) * num_tiles // gmaps_tile_size\n",
+    "    sin_y = math.sin(lat * (math.pi / 180.0))  \n",
+    "    y_coord = ((gmaps_tile_size / 2) + 0.5 * math.log((1 + sin_y) / (1 - sin_y)) * \n",
+    "               -(gmaps_tile_size / (2 * math.pi))) * num_tiles // gmaps_tile_size\n",
```

Comment:
That's a lambert equal-aread projection? Or is it mercator? Maybe we should put that outside as its own function.

Reply:
No idea, I'm going to look into it a bit deeper..

Reply:
updated with separate function and link to the formula with explanation

Reply:
forgot to say, it's mercator :globe_with_meridians: